src/applications/personalization/profile360/containers/CopyMailingAddress.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import isEqual from 'lodash/isEqual';
+import pick from 'lodash/pick';
+
+import {
+  consolidateAddress
+} from '../../../../platform/forms/address/helpers';
+
+import {
+  FIELD_NAMES
+} from '../constants/vet360';
+
+import {
+  selectVet360Field,
+  selectEditedFormField
+} from '../selectors';
+
+import {
+  cleanAddressDataForUpdate
+} from '../actions/misc';
+
+class CopyMailingAddress extends React.Component {
+  onChange = (event) => {
+    event.stopPropagation();
+    if (event.target.checked) {
+      this.props.copyMailingAddress(
+        consolidateAddress(this.props.mailingAddress)
+      );
+    }
+  }
+
+  render() {
+    if (this.props.hasEmptyMailingAddress) return <div/>;
+    return (
+      <div className="copy-mailing-address-to-residential-address">
+        <div className="form-checkbox-buttons">
+          <input
+            type="checkbox"
+            name="copy-mailing-address-to-residential-address"
+            id="copy-mailing-address-to-residential-address"
+            autoComplete="false"
+            checked={this.props.checked}
+            onChange={this.onChange}/>
+          <label htmlFor="copy-mailing-address-to-residential-address">
+            My home address is the same as my mailing address.
+          </label>
+        </div>
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  const addressProps = [
+    'addressLine1',
+    'addressLine2',
+    'addressLine3',
+    'addressPou',
+    'addressType',
+    'city',
+    'countryName',
+    'internationalPostalCode',
+    'province',
+    'stateCode',
+    'zipCode'
+  ];
+
+  const mailingAddress = selectVet360Field(state, FIELD_NAMES.MAILING_ADDRESS);
+  const hasEmptyMailingAddress = !mailingAddress;
+
+  const residentialAddress = selectEditedFormField(state, FIELD_NAMES.RESIDENTIAL_ADDRESS).value;
+
+  const mailingAddressConsolidated = pick(cleanAddressDataForUpdate(consolidateAddress(mailingAddress)), addressProps);
+  const residentialAddressConsolidated = pick(consolidateAddress(residentialAddress), addressProps);
+
```

Comment:
This is a crazy amount of conversions. We need to revisit this when we clean up the address-formatting logic.

Reply:
Yep; the address format has 3 forms that we need to deal with
1. natural address shape from the server (stored as-is in the user object)
2. formatted address shape for rendering in the edit modal
3. stripped/cleaned up address shape for sending to Vet360 for an update. we will get rejections if there are any extraneous attributes/inputs that are unexpected